apps/dapp/src/components/Pages/Core/NewUI/ClaimModal.tsx

```diff
@@ -1,31 +1,209 @@
 import { Popover } from 'components/Popover';
 import styled from 'styled-components';
-import { Button } from 'components/Button/Button';
+import { useVaultContext } from '../VaultContext';
+import { formatBigNumber, formatTemple } from 'components/Vault/utils';
+import { format, isDate } from 'date-fns';
+import { ZERO } from 'utils/bigNumber';
+import { useEffect, useState } from 'react';
+import { Vault, VaultGroup } from 'components/Vault/types';
+import { BigNumber } from 'ethers';
+import { useWithdrawFromVault } from 'hooks/core/use-withdraw-from-vault';
+import { VaultButton } from '../VaultPages/VaultContent';
+import { useTokenContractAllowance } from 'hooks/core/use-token-contract-allowance';
+import { VaultBalance } from 'hooks/core/use-vault-group-token-balance';
+import env from 'constants/env';
+import _ from 'lodash';
+import { ContinuousColorLegend } from 'react-vis';
 
 interface IProps {
   isOpen: boolean;
   onClose: () => void;
 }
 
+const EMPTY_CLAIM_STATE = {
+  claimSubvaultAddress: '',
+  isEarly: false,
+  claimAmount: '',
+};
+
 export const ClaimModal: React.FC<IProps> = ({ isOpen, onClose }) => {
+  const {
+    balances: { balances, isLoading: balancesIsLoading },
+    vaultGroups: { vaultGroups, isLoading: vaultGroupsIsLoading },
+    refreshVaultBalance,
+  } = useVaultContext();
+
+  const { withdrawFromVault: withdrawRequest, withdrawEarly: earlyWithdrawRequest } = useWithdrawFromVault(
+    '',
+    async () => {
+      await refreshVaultBalance(claimState.claimSubvaultAddress);
+      // AnalyticsService.captureEvent(AnalyticsEvent.Vault.Claim, { name: vault.id, amount });
+      clearClaimState();
+    }
+  );
+
+  const [withdrawFromVault, { isLoading: withdrawIsLoading, error: withdrawError }] = withdrawRequest;
+  const [earlyWithdraw, { isLoading: earlyWithdrawIsLoading, error: earlyWithdrawError }] = earlyWithdrawRequest;
+
+  const [assumedActiveVaultGroup, setAssumedActiveVaultGroup] = useState({} as VaultGroup);
+
+  useEffect(() => {
+    if (!balancesIsLoading && !vaultGroupsIsLoading) {
+      setAssumedActiveVaultGroup(vaultGroups[0]);
+      const nonZeroBalances = getVaultBalances(vaultGroups[0]?.vaults).filter(vb => vb);
+      setZeroVaultBalance(nonZeroBalances.length === 0);
+    }
+  }, [balancesIsLoading, vaultGroupsIsLoading, vaultGroups]);
+
+  const [claimState, setClaimState] = useState(EMPTY_CLAIM_STATE);
+  const [zeroVaultBalance, setZeroVaultBalance] = useState(false);
+
+  const claimAmountHandler = (contract: string, value: BigNumber, isEarly: boolean) => {
+    setClaimState({
+      claimSubvaultAddress: contract,
+      // TODO: Update with the actual amount, because I don't want to withdraw everything because I can't re-deposit
```

Comment:
update this

Reply:
https://github.com/TempleDAO/temple/pull/701/commits/fbf1a0a04fdda944194a421d727178292d459591